sdcflows/interfaces/bspline.py

```diff
@@ -0,0 +1,387 @@
+        fmap_points = apply_affine(fmapnii.affine.astype("float32"), mask_indices)
+
+        # Calculate the spatial location of control points
+        bs_levels = []
+        w_l = []
+        ncoeff = []
+        for sp in bs_spacing:
+            level = bspline_grid(fmapnii, control_zooms_mm=sp)
+            bs_levels.append(level)
+            ncoeff.append(level.dataobj.size)
+            w_l.append(bspline_weights(fmap_points, level))
+
+        # Compose the interpolation matrix
+        regressors = np.vstack(w_l)
+
+        # Fit the model
+        model = lm.Ridge(alpha=self.inputs.ridge_alpha, fit_intercept=False)
+        model.fit(regressors.T, data[mask])
+
+        interp_data = np.zeros_like(data)
+        interp_data[mask] = np.array(model.coef_) @ regressors  # Interpolation
+
+        # Store outputs
+        out_name = fname_presuffix(
+            self.inputs.in_data, suffix="_field", newpath=runtime.cwd
+        )
+        hdr = fmapnii.header.copy()
+        hdr.set_data_dtype("float32")
+        nb.Nifti1Image(interp_data, fmapnii.affine, hdr).to_filename(out_name)
+        self._results["out_field"] = out_name
+
+        index = 0
+        self._results["out_coeff"] = []
+        for i, (n, bsl) in enumerate(zip(ncoeff, bs_levels)):
+            out_level = out_name.replace("_field.", f"_coeff{i:03}.")
+            nb.Nifti1Image(
+                np.array(model.coef_, dtype="float32")[index:index + n].reshape(
+                    bsl.shape
+                ),
+                bsl.affine,
+                bsl.header,
+            ).to_filename(out_level)
+            index += n
+            self._results["out_coeff"].append(out_level)
+
+        # Write out fitting-error map
+        self._results["out_error"] = out_name.replace("_field.", "_error.")
+        nb.Nifti1Image(
+            data * mask - interp_data, fmapnii.affine, fmapnii.header
+        ).to_filename(self._results["out_error"])
+
+        if not self.inputs.extrapolate:
+            return runtime
+
+        bg_indices = np.argwhere(~mask)
+        bg_points = apply_affine(fmapnii.affine.astype("float32"), bg_indices)
+
+        extrapolators = np.vstack(
+            [bspline_weights(bg_points, level) for level in bs_levels]
+        )
+        interp_data[~mask] = np.array(model.coef_) @ extrapolators  # Extrapolation
+        self._results["out_extrapolated"] = out_name.replace("_field.", "_extra.")
+        nb.Nifti1Image(interp_data, fmapnii.affine, hdr).to_filename(
+            self._results["out_extrapolated"]
+        )
+        return runtime
+
+
+class _Coefficients2WarpInputSpec(BaseInterfaceInputSpec):
+    in_target = File(exist=True, mandatory=True, desc="input EPI data to be corrected")
+    in_coeff = InputMultiObject(
+        File(exists=True),
+        mandatory=True,
+        desc="input coefficients, after alignment to the EPI data",
+    )
+    ro_time = traits.Float(1.0, usedefault=True, desc="EPI readout time (s).")
```

Comment:
i'm a bit concerned with setting this by default

Reply:
```suggestion
    ro_time = traits.Float(mandatory=True, desc="EPI readout time (s).")
```